### PR TITLE
Fix create ort value hardcoded memory info to CPU

### PR DESCRIPTION
**Pull request by anhnguyen7198**
In CreateMLValue [here](https://github.com/microsoft/onnxruntime/blob/7e5d68eea664e362c99c298e8747a8b7314dab8b/orttraining/orttraining/eager/ort_util.cpp#L28), the OrtMemoryInfo of the Tensor being created is hardcoded to CPU memory, this PR fix the hardcode by retrieving the OrtMemoryInfo directly from the Allocator

### CODE REVIEW

File: orttraining/orttraining/eager/ort_aten.cpp [hunk 1]
```diff
@@ -130,11 +132,26 @@ OrtValue create_ort_value(
   }
 
   OrtValue ort_tensor;
-  CreateMLValue(
-    tensor.data_ptr(),
-    ort_scalar_type_from_aten(tensor.scalar_type()),
-    tensor.sizes().vec(),
-    &ort_tensor);
+  if (tensor.device().type() == at::kORT) {
```
Comment (souptc) [inline orttraining/orttraining/eager/ort_aten.cpp:135]: if input is a tensor on ORT device, it should meet the condition at line 129. why you need this?

Comment (anhnguyen7198) [reply orttraining/orttraining/eager/ort_aten.cpp:135]: not sure I fully understand but what you mean is if tensor is on ORT device, then these lines will be triggered, right?

```
 auto* impl = dynamic_cast<ORTTensorImpl*>(tensor.unsafeGetTensorImpl());
  if (impl) {
    return impl->tensor();
  }
```

File: orttraining/orttraining/eager/ort_aten.cpp [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 #include "ort_aten.h"
 #include "ort_tensor.h"
+#include "ort_backends.h"
+#include <core/providers/cpu/cpu_execution_provider.h>
```
Comment (souptc) [inline orttraining/orttraining/eager/ort_aten.cpp:7]: why we need those headers?

Comment (anhnguyen7198) [reply orttraining/orttraining/eager/ort_aten.cpp:7]: I think we need those headers to create OrtMemoryInfo

`  OrtMemoryInfo *mem_info;
  Ort::ThrowOnError(Ort::GetApi().CreateCpuMemoryInfo(OrtArenaAllocator, OrtMemTypeDefault, &mem_info));`

Comment (anhnguyen7198) [reply orttraining/orttraining/eager/ort_aten.cpp]: Fixed

**Review by souptc — APPROVED** [review]

**Pull request merged**